boot/boot.go

```diff
@@ -360,5 +360,61 @@ func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet) e
 	}
 
 	return nil
+}
+
+func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet, opts *bootloader.Options) error {
+	// we can only make a single recovery system bootable right now
+	recoverySystems, err := filepath.Glob(filepath.Join(rootdir, "systems/*"))
+	if err != nil {
+		return fmt.Errorf("cannot validate recovery systems: %v", err)
+	}
+	if len(recoverySystems) > 1 {
+		return fmt.Errorf("cannot make multiple recovery systems bootable yet")
+	}
+
+	// install the bootloader configuration from the gadget
+	if err := bootloader.InstallBootConfig(bootWith.UnpackedGadgetDir, rootdir, opts); err != nil {
+		return err
+	}
+
+	// TODO:UC20: extract kernel for e.g. ARM
+
+	// now install the recovery system specific boot config
+	bl, err := bootloader.Find(rootdir, opts)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot find bootloader: %v", err)
+	}
+	rbl, ok := bl.(bootloader.RecoveryAwareBootloader)
+	if !ok {
+		return fmt.Errorf("cannot use %s bootloader: does not support recovery systems", bl.Name())
+	}
+	kernelPath, err := filepath.Rel(rootdir, bootWith.KernelPath)
+	if err != nil {
+		return fmt.Errorf("cannot construct kernel boot path: %v", err)
+	}
+	blVars := map[string]string{
+		"snapd_recovery_kernel": filepath.Join("/", kernelPath),
```

Comment:
https://github.com/snapcore/pc-amd64-gadget/blob/20/grub.cfg-recovery#L57 <- this line needs to change once this lands. Instead of hardcoding the /snaps prefix now the bootenv will set the right full path.